dex/src/main/java/io/crate/data/join/BlockSizeCalculator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.data.join;
+
+/**
+ * Calculates the best size for a block of Rows in memory.
+ */
+public interface BlockSizeCalculator {
```

Comment:
What's wrong with sticking to the existing `IntSupplier`?

Reply:
I find `BlockSizeCalculator` conveys the meaning much better than the generic `IntSupplier`. We might also add parameters or other methods to the interface in the future.

I know it's an extra interface but I prefer the increased readability.

Reply:
Then I'd appreciate to at least put this change into a separate commit as it's not strictly necessary.

> We might also add parameters or other methods to the interface in the future.

Anticipating future needs is generally not the best thing to do. More often than not the anticipations turn out to be wrong and just overcomplicate things.

I don't really see how `BlockSizeCalculator blockSizeCalculator` is more readable than `IntSupplier blockSizeCalculator`. I'd think that the later is easier to read because java devs are already familiar with an `IntSupplier`. In the first case they've to take a look at the interface to see what it does. But given that it is a minor thing that doesn't really matter I leave it up to you.

Reply:
I knew when writing you would bring the future-proof counterargument :) How is this interface over-complicating things? I understand your argument for other cases but don't see how it harms here. I don't mind removing it.